src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProvider.cs

```diff
@@ -178,18 +181,43 @@ bool TryAddCompletion(string attributeName, BoundAttributeDescriptor boundAttrib
 
             void AddCompletion(string attributeName, BoundAttributeDescriptor boundAttributeDescriptor, TagHelperDescriptor tagHelperDescriptor)
             {
-                if (!attributeCompletions.TryGetValue(attributeName, out var attributeDescriptionInfos))
+                if (!attributeCompletions.TryGetValue(attributeName, out var attributeDetails))
                 {
-                    attributeDescriptionInfos = new HashSet<AttributeDescriptionInfo>();
-                    attributeCompletions[attributeName] = attributeDescriptionInfos;
+                    attributeDetails = (new HashSet<AttributeDescriptionInfo>(), new HashSet<string>());
+                    attributeCompletions[attributeName] = attributeDetails;
                 }
 
+                (var attributeDescriptionInfos, var commitCharacters) = attributeDetails;
+
                 var descriptionInfo = new AttributeDescriptionInfo(
                     boundAttributeDescriptor.TypeName,
                     tagHelperDescriptor.GetTypeName(),
                     boundAttributeDescriptor.GetPropertyName(),
                     boundAttributeDescriptor.Documentation);
                 attributeDescriptionInfos.Add(descriptionInfo);
+
+                if (attributeName.EndsWith("..."))
+                {
+                    // Indexer attribute, we don't want to commit with standard chars
+                    return;
+                }
+
+                if (!commitCharacters.Contains("="))
```

Comment:
`commitCharacters` is a `HashSet`,  You can always add the `=` character and if it already exists it'll be ignored.